src/components/FAB.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import {Pressable, Animated, Easing} from 'react-native';
+import PropTypes from 'prop-types';
+import {PlusIcon} from './Expensicons';
+import styles from '../styles/styles';
+import themeColors from '../styles/themes/default';
+
+const AnimatedPlusIcon = Animated.createAnimatedComponent(PlusIcon);
+const AnimatedPressable = Animated.createAnimatedComponent(Pressable);
+
+const propTypes = {
+    // Callback to fire on request to toggle the FAB
+    onPress: PropTypes.func.isRequired,
+
+    // Current state (active or not active) of the component
+    isActive: PropTypes.bool.isRequired,
+
+    // Temporary prop to be able to test the component before implementing it
+    isHidden: PropTypes.bool,
+};
+
+const defaultProps = {
+    isHidden: true,
+};
+
+class FAB extends React.Component {
+    constructor(props) {
+        super(props);
+        this.animatedValue = new Animated.Value(0);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.isActive !== this.props.isActive) {
+            this.animateFloatinActionButton();
+        }
+    }
+
+    animateFloatinActionButton() {
```

Comment:
We require adding JSDocs to all class methods just a description of the method is OK since there is no param or return value.

Reply:
Added the description.